src/ALE/MOM_remapping.F90

```diff
@@ -1415,387 +1670,794 @@ logical function remapping_unit_tests(verbose)
+    this%verbose = verbose
+  endif
+  if (present(stdout)) then
+    this%stdout = stdout
+  endif
+  if (present(stderr)) then
+    this%stderr = stderr
+  endif
+  if (present(stop_instantly)) then
+    this%stop_instantly = stop_instantly
+  endif
+end subroutine set
+
+!> Returns state
+logical function outcome(this)
+  class(testing), intent(inout) :: this !< This testing class
+  outcome = this%state
+end function outcome
+
+!> Summarize results
+logical function summarize(this, label)
+  class(testing),  intent(inout) :: this  !< This testing class
+  character(len=*),   intent(in) :: label !< Message
+  integer :: i
+
+  if (this%state) then
+    write(this%stdout,'(a," : ",a,", ",i4," failed of ",i4," tested")') &
+         red('FAIL'), trim(label), this%num_tests_failed, this%num_tests_checked
+    write(this%stdout,'(a,100i4)') 'Failed tests:',(this%ifailed(i),i=1,this%num_tests_failed)
+    write(this%stdout,'(a,a)') 'First failed test: ',trim(this%label_first_fail)
+    write(this%stderr,'(a,100i4)') 'Failed tests:',(this%ifailed(i),i=1,this%num_tests_failed)
+    write(this%stderr,'(a,a)') 'First failed test: ',trim(this%label_first_fail)
+    write(this%stderr,'(a," : ",a)') trim(label),'FAILED'
+  else
+    write(this%stdout,'(a," : ",a,", all ",i4," tests passed")') &
+         green('Pass'), trim(label), this%num_tests_checked
+  endif
+  summarize = this%state
+end function summarize
+
+!> Pads string with the color codes for red/reset
+function red(string) result(newstr)
+  character(len=*),  intent(in) :: string !< Input string
+  character(len=:), allocatable :: newstr !< The modified output string
+  newstr = achar(27)//'[31m'//trim(string)//achar(27)//'[0m'
+end function red
+
+!> Pads string with the color codes for green/reset
+function green(string) result(newstr)
+  character(len=*),  intent(in) :: string !< Input string
+  character(len=:), allocatable :: newstr !< The modified output string
+  newstr = achar(27)//'[32m'//trim(string)//achar(27)//'[0m'
+end function green
+
+!> Compare u_test to u_true, report, and return true if a difference larger than tol is measured
+!!
+!! If in verbose mode, display results to stdout
+!! If a difference is measured, display results to stdout and stderr
+subroutine real_arr(this, n, u_test, u_true, label, tol)
+  class(testing),  intent(inout) :: this   !< This testing class
   integer,            intent(in) :: n      !< Number of cells in u
-  real, dimension(n), intent(in) :: u      !< Values to test [A]
+  real, dimension(n), intent(in) :: u_test !< Values to test [A]
   real, dimension(n), intent(in) :: u_true !< Values to test against (correct answer) [A]
   character(len=*),   intent(in) :: label  !< Message
   real,     optional, intent(in) :: tol    !< The tolerance for differences between u and u_true [A]
   ! Local variables
-  real :: tolerance ! The tolerance for differences between u and u_true [A]
   integer :: k
+  logical :: this_test
+  real :: tolerance, err
```

Comment:
Please add a description of these real variables (`tolerance` and `err`), including their units.